### DISCUSSION TITLE
Full neumann BC or pressure: comparisons

### DISCUSSION THREAD

**Discussion by pescap** [lululxvi/deepxde#950] [Q&A]
Dear all, In the [Beltrami flow code](https://github.com/lululxvi/deepxde/blob/master/examples/pinn_forward/Beltrami_flow.py), the pressure is defined up to a constant.

In order to compare the predicted and exact pressions, I was trying to find a way to realize a proper shifting. 

In [NSFnets (Navier-Stokes Flow nets): Physics-informed neural networks for
the incompressible Navier-Stokes equations](https://www.sciencedirect.com/science/article/pii/S0021999120307257), p.11, the authors: "apply a shift for the NSFnet simulation results to bring the means of the pressure for reference DNS results and NSFnet results to the same value."

In FEM simulations, e.g. for Neumann BCs, one often adds a term to the variational formulation (ref: [Steinbach](https://link.springer.com/book/10.1007/978-0-387-68805-3)).

A few ideas:
- Evaluate analytically the mean value of the exact solution over `D \times [0,T]` and remove it, to obtain an exact solution with zero average ("centered").
- Apply a transform to the NN to obtained a centered pressure
- Modify the loss function to obtain a centered pressure

Do you know how to proceed? 
Best,

**Comment by lululxvi** [comment]
All sound good. But I think the easiest way is what was done in NSFnet, i.e., after training, do a post-processing to shift the pressure.
